debugtools/DDR_VM/src/com/ibm/j9ddr/tools/ddrinteractive/CacheMemory.java

```diff
@@ -0,0 +1,128 @@
+ * Secondary Licenses when the conditions for such availability set                                                                                                          
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU                                                                                                            
+ * General Public License, version 2 with the GNU Classpath                                                                                                                  
+ * Exception [1] and GNU General Public License, version 2 with the                                                                                                          
+ * OpenJDK Assembly Exception [2].                                                                                                                                           
+ *                                                                                                                                                                           
+ * [1] https://www.gnu.org/software/classpath/license.html                                                                                                                   
+ * [2] http://openjdk.java.net/legal/assembly-exception.html                                                                                                                 
+ *                                                                                                                                                                           
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception                                                                                                                                                                                                                
+ *******************************************************************************/
+
+package com.ibm.j9ddr.tools.ddrinteractive;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.nio.ByteOrder;
+import java.util.Properties;
+
+import com.ibm.j9ddr.corereaders.memory.AbstractMemory;
+import com.ibm.j9ddr.corereaders.memory.IAddressSpace;
+import com.ibm.j9ddr.corereaders.Platform;
+import com.ibm.j9ddr.corereaders.ICore;
+import com.ibm.j9ddr.corereaders.memory.IModule;
+import com.ibm.j9ddr.corereaders.memory.IProcess;
+import com.ibm.j9ddr.corereaders.memory.IMemorySource;
+import com.ibm.j9ddr.corereaders.memory.MemoryFault;
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.DataUnavailableException;
+import com.ibm.j9ddr.corereaders.osthread.IOSThread;
+
+
+public class CacheMemory extends AbstractMemory implements IProcess
+{
+
+    public CacheMemory(ByteOrder byteOrder){
```

Comment:
Please reformat:
* tabs are 4 positions (to be used only for leading whitespace)
* braces should be preceded by a space
* collapse multiple blank lines to one
* one blank line before each method

Add javadoc to explain the class and its purpose

Add missing `@Override` annotations

Reply:
will do, also agree.

Also related to purpose: what is your opinion on how the base address should be provided? 
in the PR comment I've linked the "feature" that I've created to expose the romclass portion of the SCC in order to get a base address, should this "feature" also be PR'd (either in its own or separate)?

Reply:
Everything in the SCC should be position-independent and so the address should be irrelevant. Given a `FileChannel` on the SCC, you can use `FileChannel.map()` to get a byte buffer - you can pretend the buffer is mapped at any (non-zero) address you like, so the changes in the commit you reference should not be necessary.

Reply:
cool, 
one more clarification - style this time - no one uses `@Override`s for interface methods right? just subclass?

Reply:
You should use `@Override` everywhere the compiler will let you; that means methods that correspond to declarations in interfaces too. Besides documentation, they have value in keeping things in sync with interface and base class changes - if you change the signature of a method, the compiler will issue errors for methods marked with `@Override` that no longer match so you know to fix them to keep the behavior you intended.

Reply:
I think `@see` has value independent of the target of such a reference.